app/uk/gov/hmrc/gform/formtemplate/MiniSummaryListHelper.scala

```diff
@@ -0,0 +1,183 @@
+    fields.fold(fields)(fs => NonEmptyList.fromList(updateMsl(fs.toList, formTemplate)))
+
+  private def updateSection(section: Section, formTemplate: FormTemplate): Section =
+    section.fold[Section](nonRepeatingPage =>
+      nonRepeatingPage.copy(page = updatePage(nonRepeatingPage.page, formTemplate))
+    )(repeatingPage => repeatingPage.copy(page = updatePage(repeatingPage.page, formTemplate)))(addToList =>
+      addToList.copy(pages = addToList.pages.map(p => updatePage(p, formTemplate)))
+    )
+
+  private def updatePage(page: Page, formTemplate: FormTemplate): Page = {
+    val fields = updateMsl(page.fields, formTemplate)
+    page.copy(fields = fields)
+  }
+
+  private def updateMsl(
+    fields: List[FormComponent],
+    formTemplate: FormTemplate
+  ): List[FormComponent] =
+    fields.map(f =>
+      f.`type` match {
+        case MiniSummaryList(rows, displayInSummary, keyDisplayWidth) =>
+          f.copy(`type` = MiniSummaryList(updateMslRows(rows, formTemplate), displayInSummary, keyDisplayWidth))
+        case _ => f
+      }
+    )
+
+  private def updateMslRows(
+    mslRows: List[MiniSummaryRow],
+    formTemplate: FormTemplate
+  ): List[MiniSummaryRow] =
+    mslRows.map {
+      case MiniSummaryRow.ValueRow(key, MiniSummaryListValue.AnyExpr(exp), includeIf, pageId, taskId) =>
+        MiniSummaryRow.ValueRow(
+          key,
+          MiniSummaryListValue.AnyExpr(exp),
+          createIncludeIf(includeIf, pageId, taskId, formTemplate),
+          pageId,
+          taskId
+        )
+      case MiniSummaryRow.ValueRow(key, r, includeIf, pageId, taskId) =>
+        MiniSummaryRow.ValueRow(
+          key,
+          r,
+          createIncludeIf(includeIf, pageId, taskId, formTemplate),
+          pageId,
+          taskId
+        )
+      case MiniSummaryRow.SmartStringRow(key, r, includeIf, pageId, taskId) =>
+        MiniSummaryRow.SmartStringRow(
+          key,
+          r,
+          createIncludeIf(includeIf, pageId, taskId, formTemplate),
+          pageId,
+          taskId
+        )
+      case header @ MiniSummaryRow.HeaderRow(_) =>
+        header
+      case MiniSummaryRow.ATLRow(atlId, includeIf, rows) =>
+        MiniSummaryRow.ATLRow(atlId, includeIf, updateMslRows(rows, formTemplate))
+    }
+
+  private def createIncludeIf(
+    rowIncludeIf: Option[IncludeIf],
+    pageIdOpt: Option[PageId],
+    taskIdOpt: Option[TaskId],
+    formTemplate: FormTemplate
+  ): Option[IncludeIf] = {
+    def combineIncludeIf(a: IncludeIf, b: IncludeIf): IncludeIf =
+      new IncludeIf(And(a.booleanExpr, b.booleanExpr))
+
+    def handlePagesParentTask(pageIncludeIf: Option[IncludeIf], taskIncludeIf: Option[IncludeIf]): Option[IncludeIf] =
+      (pageIncludeIf, taskIncludeIf) match {
+        case (Some(pageIf), Some(taskIf)) => Option(combineIncludeIf(pageIf, taskIf))
+        case (None, Some(taskIf))         => Option(taskIf)
+        case (Some(pageIf), None)         => Option(pageIf)
```

Comment:
Let's not use an `Option` when we are not dealing with `null`s. 

It this case `Some` is perfectly fine.